### PR TITLE
master list, list view

### LINKED ISSUE #801: Master lists: List
Show a list of master lists for the currently logged in store

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #801 

Adds a list view for master lists. had some trouble typing the mock data response.. and, as you'll see, gave up.

### CODE REVIEW

File: packages/config/src/routes.ts [hunk 1]
```diff
@@ -14,8 +14,9 @@ export enum AppRoute {
 
   Inventory = 'inventory',
   Stock = 'stock',
-  Stocktake = 'stocktake',
+  Stocktakes = 'stocktakes',
```
Comment (mark-prins) [inline packages/config/src/routes.ts:17]: a little pedantic.. there was `Locations` `Items` `Requisitions` etc, but the `Stocktake` was singular.

File: packages/mock-server/src/data/data.ts [hunk 2]
```diff
@@ -747,15 +750,50 @@ export const createStocktakeLine = (
...
-      const stockLineSubset = takeRandomSubsetFrom(stockLines, 100);
+      const itemsSubset = takeRandomSubsetFrom(ItemData, 50);
+      return itemsSubset.map(item => createStocktakeLine(stocktake.id, item));
+    })
+    .flat();
+};
+
+const createMasterList = (n: number): MasterList => ({
+  id: faker.datatype.uuid(),
+  code: `LIST${n}`,
```
Comment (mark-prins) [inline packages/mock-server/src/data/data.ts:765]: was thinking to make some static data, in the end I didn't and made some crappy data instead.

File: packages/mock-server/src/worker/handlers/masterList/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { mockMasterListsQuery } from '@openmsupply-client/common/src/types';
+import { ResolverService } from '../../../api/resolvers';
+
+const masterListsQuery = mockMasterListsQuery((req, res, ctx) => {
+  return res(
+    ctx.data({
+      // trouble typing the response correctly.. gave up. it's only mocks after all
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      masterLists: ResolverService.masterList.list(req?.variables) as any,
```
Comment (mark-prins) [inline packages/mock-server/src/worker/handlers/masterList/index.ts:9]: this gave me grief.. decided eventually that I didn't care enough to spend more time on it

Comment (joshxg) [reply packages/mock-server/src/worker/handlers/masterList/index.ts:9]: Yeah, super annoying!

File: packages/system/src/MasterList/ListView/api.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+const masterListsGuard = (masterListsQuery: MasterListsQuery) => {
+  if (masterListsQuery.masterLists.__typename === 'MasterListConnector') {
+    return masterListsQuery.masterLists;
+  }
+
+  if (masterListsQuery.masterLists.__typename === 'ConnectorError') {
+    throw new Error(masterListsQuery.masterLists.error.description);
+  }
+
+  throw new Error('oh dear');
```
Comment (joshxg) [inline packages/system/src/MasterList/ListView/api.ts:20]: The `__typename` field is not optional, so this isn't needed. Sometimes it's optional and I can't really determine why that is, but when it is optional, typescript complains and this throw is required.

**Review by joshxg — APPROVED** [review]
Nice!

File: packages/system/src/MasterList/ListView/api.ts (hunk 4 above)
Comment (mark-prins) [reply packages/system/src/MasterList/ListView/api.ts:20]: thanks - I was trying to find why it was sometimes optional; the error conditions are a mix of required and optional and that bugged me 🤷 
For this one - ts was complaining.. but now is fine?? have removed it.

**Pull request merged**